core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -248,4 +254,123 @@ private boolean isPreferredSubnetsSpecifiedForScaling(NetworkScaleDetails networ
         return networkScaleDetails != null
                 && CollectionUtils.isNotEmpty(networkScaleDetails.getPreferredSubnetIds());
     }
+
+    public Set<GroupPlacement> prepareGroupPlacementSet(Map<String, String> subnetAzPairs, Set<InstanceGroup> instanceGroups) {
+        Set<String> hints = getHints(instanceGroups);
+        Set<String> availabilityZones = getAvailabilityZones(subnetAzPairs);
+
+        if (availabilityZones.size() < ALLOWED_NUMBER_OF_HINTS || hints.size() != ALLOWED_NUMBER_OF_HINTS) {
+            return Collections.emptySet();
+        }
```

Comment:
I don't quite understand this restriction.
Why do we expect that the hint size matches the max number?

Reply:
You are right, I improved this condition and constant name. 
The number of hints cannot be smaller than 3 and the number of AZs cannot be smaller than the number of hints. In this case, it will fall back to the round-robin approach.